ants_test.go

```diff
@@ -427,24 +427,26 @@ func TestMaxBlockingSubmit(t *testing.T) {
 
 func TestNonblockingSubmitWithFunc(t *testing.T) {
 	poolSize := 10
-	ch1 := make(chan struct{})
+	var wg sync.WaitGroup
 	p, err := NewPoolWithFunc(poolSize, func(i interface{}) {
 		longRunningPoolFunc(i)
-		close(ch1)
+		wg.Done()
 	}, WithNonblocking(true))
 	assert.NoError(t, err, "create TimingPool failed: %v", err)
 	defer p.Release()
+	ch := make(chan struct{})
+	wg.Add(poolSize-1)
```

Comment:
Sorry for not giving a thoughtful comment, please just change this to `wg.Add(poolSize)` and remove the below one `wg.Add(1)`.

Reply:
done. either way, it's all equivalent.